lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1591,20 +1591,10 @@ def __init__(
             if not isinstance(refresh_event, list):
                 refresh_event = [refresh_event]
 
+        self.framework.observe(events.relation_joined, self.set_scrape_job_spec)
         for ev in refresh_event:
             self.framework.observe(ev, self.set_scrape_job_spec)
 
-        # Update relation data every reinit. If instead we used event hooks then observing only
-        # relation-joined would not be sufficient:
-        # - Would need to observe leader-elected, in case there was no leader during
-        #   relation-joined.
-        # - If later related to an ingress provider, then would need to register and wait for
```

Comment:
Would you mind clarifying how the ingress code ordering problem is no longer an issue?

Reply:
Because it was corrected [here](https://github.com/canonical/prometheus-k8s-operator/pull/397) and [here](https://github.com/canonical/prometheus-k8s-operator/pull/405), so that's finished. Even if it were not, it can't be done this way in the long run, and would require a separate patch.

Reply:
OK yeah with #397 it's not ideal but should work. (#405 seems unrelated here.)